src/lib/gundb/gundb-extend.js

```diff
@@ -71,9 +71,51 @@ assign(chain, {
       .then(callback || identity)
       .catch(noop)
   },
+
+  /**
+   * Returns the decrypted value
+   * @returns {Promise<any>}
+   */
+  async decrypt(callback = null) {
```

Comment:
1. code repetition here. please move this to the separate function `export const decrypt = async (node, callback = null)` in the gun utils and call from here as `return await decrypt(this, callback)`
2. this caches secure key value. if you'll .secretAck again, you will try to decrypt with the old secureKey value then and will get an error. to avoid this, you need override `.secretAck` also and call `flushSecureKey(this)` after encryption done (like i did in the Gun.user's secretAck override)

Reply:
1. we can remove decrypt from Gun.user as now every node has decrypt
2. i dont understand where we need to add flush, please explain with an example

Reply:
@sirpy 

1.  gun.get('somenode').secretAck('somevalue')
2. gun.get('somenode').decrypt() - here we getting secure key and caching it by the node object via WeakMap
3. gun.get('somenode').secretAck('othervalue') - here we have new secure key generated. but the cache still have the old one
4. gun.get('somenode').decrypt() - will got an error because the old secure key will be used

To avoid this case i added flushSecureKey(this) to the secretAck override of the Gun.user.
As you've made the override global, for each node, you have move secretAck override from Gun.user to Gun.chain too

Reply:
1. secretAck only generates a key if it doesnt exists, so i'm not sure flush is also needed in your implementation
2. secretAck should stay on Gun.user as it calls "this.secret" which is a gun.user only method

Reply:
@sirpy please re check pt1 to make sure flush isn't needed. if it's not needed you could remove it from secrectAck and also remove flushSecureKey fn